lib/spike.rb

```diff
@@ -213,120 +102,126 @@ def option_overrides
+      return "The ID of the #{param.name.gsub("_id", "").gsub("_", " ")}" if param.name.end_with? "_id"
+      return param.description.gsub("\n", "") unless param.description.empty?
 
-      "The ID of the #{parameter.name.gsub("_id", "").gsub("_", " ")}" if parameter.name.end_with?("_id")
+      "The ID of the #{param.name.gsub("_id", "").gsub("_", " ")}" if name.end_with?("_id")
     end
 
     def parameter_documentation
-      definition.params.select(&:required).map {|param|
+      required_params.map {|param|
         "@param #{param.name} #{parameter_type(param)} #{parameter_description(param)}"
-      } + definition.params.reject(&:required).reject {|param| ["per_page", "page"].include?(param.name)}.map {|param|
-        "@param options [#{param.type.capitalize}] :#{param.name} #{param.description}"
+      } + optional_params.map {|param|
+        "@param options [#{param.type.capitalize}] :#{param.name} #{param.description.gsub("\n", "")}"
       }
     end
 
     def return_value_description
       case verb
       when "GET"
-        if resource.singular?
-          "A single #{resource.name.gsub("_", " ")}"
+        if singular?
+          "A single #{namespace.gsub("_", " ")}"
         else
-          "A list of #{resource.name.gsub("_", " ")}"
+          "A list of #{namespace.gsub("_", " ")}"
         end
       when "POST"
-        "The new #{resource.name.singularize.gsub("_", " ")}"
+        "The new #{namespace.singularize.gsub("_", " ")}"
       else
       end
     end
 
     def verb
-      definition["method"]
+      definition.method.upcase
     end
 
     def parameters
-      parameterizer.parameterize(arguments)
-    end
-
-    def arguments
-      definition.params.select(&:required).map(&:name)
+      parameterizer.parameterize(required_params.map(&:name))
     end
 
-    def resource
-      return @resource if defined?(@resource)
-
-      final_segment = path_segments.reject {|segment| segment.start_with?(":")}.last
-      if final_segment == directory
-        @resource = final_segment
-      else
-        @resource = [directory.singularize, final_segment].join("_")
-      end
-    end
-
-    def path_segments
-      @path_segments ||= definition.path.to_s.split("/").reject(&:empty?)
+    def namespace
+      definition.operation_id.split("/").last.split("-").drop(1).join("_")
     end
 
     def method_name
       case verb
       when "GET"
-        resource.name
+        namespace
       when "POST"
-        "create_#{resource.name.singularize}"
+        # "create_#{namespace.singularize}"
+        "create_#{namespace.gsub("-", "_")}"
       else
       end
     end
 
     def alternate_name
-      return unless definition.verb == "GET"
-      return if resource.singular?
-      "list_#{resource.name}"
+      return unless verb == "GET"
+      return if singular?
+      "list_#{namespace}"
     end
 
+    # TODO. `priority` is calculating the wrong priorities bc of this method
```

Comment:
I would appreciate some 👀 here as I'm having trouble figuring out what this should be calculating for the next method `priority` to work correctly